confounds/tests/test_confounds.py

```diff
@@ -62,6 +63,27 @@ def test_residualize_linear():
             assert_almost_equal(residual_train_X.T.dot(train_confounds), 0)
 
 
+def test_residualize_targets_linear():
+    """sanity checks on implementation"""
+
+    min_dim = 6  # atleast 4+ required for make_sparse_uncorrelated
+    max_dim = 100
+    for n_samples in np.random.randint(0, 20, 1):
+        for num_confounds in np.random.randint(min_dim, max_dim, 3):
+            train_all, _ = make_sparse_uncorrelated(
```

Comment:
this does not generate discrete numerical/integer values! We need a way to broaden the test with multiple data types!

Reply:
a bit confused here, does this look better? `np.random.randint(0, 20, 1, dtype=int)`

Reply:
before updating the tests, how about you try making a jupyter notebook, and play with this class, and write some examples, to see how it works?

Reply:
hey @raamana sorry for the late response. this summer has been a bit crazy. probably need more time-out here. i noticed that some repos have been snapshotted in Arctic Code Vault, i guess it didn't snapshot PR  :p

Reply:
NP, take your time. Happy to chat with you to discuss this and make a more precise plan to finish it quickly.

why are you referencing the Arctic Code Vault? did you lose your fork or what?

Reply:
thanks! i'm wondering if the snapshot in Arctic Code Vault would include PR? and no, i didn't lose forks.

Reply:
I don't know - but I am sure your contribution will be clearly recognized :)

Reply:
hi @zuxfoucault, would you have time this year to work on this?

Reply:
Hi, sorry I just saw this.